File: app/src/main/java/com/wolcano/musicplayer/music/ui/adapter/other/CopyItemAdapter.java
package com.wolcano.musicplayer.music.ui.adapter.other;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.wolcano.musicplayer.music.R;
import com.wolcano.musicplayer.music.databinding.ItemCopyBinding;
import com.wolcano.musicplayer.music.mvp.models.Copy;

import java.util.ArrayList;


public class CopyItemAdapter extends RecyclerView.Adapter<CopyItemAdapter.ViewHolder> {

    private final ArrayList<Copy> copyList;
    private final Context context;
    private ItemCallback itemCallback;

    public CopyItemAdapter(Context context, ArrayList<Copy> copyList) {
        this.context = context;
        this.copyList = copyList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemCopyBinding binding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.item_copy, parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.binding.setCopy(copyList.get(position));
        holder.binding.executePendingBindings();
        Copy copy = holder.binding.getCopy();
        holder.binding.text.setText(copy.getText());
        if (copy.getIcon() == 0)
            holder.binding.icon.setImageDrawable(context.getResources().getDrawable(R.drawable.baseline_music_note_white_24));
        if (copy.getIcon() == 1)
            holder.binding.icon.setImageDrawable(context.getResources().getDrawable(R.drawable.account_circle_white));
        if (copy.getIcon() == 2)
            holder.binding.icon.setImageDrawable(context.getResources().getDrawable(R.drawable.baseline_album_white_36));

    }

    @Override
    public int getItemCount() {
        return copyList.size();
    }

    public void setCallback(ItemCallback callback) {
        this.itemCallback = callback;
    }

    public interface ItemCallback {

        void onItemClicked(int itemIndex);
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        private ItemCopyBinding binding;

        public ViewHolder(ItemCopyBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
            this.binding.getRoot().setOnClickListener(this::onClick);
        }

        @Override
        public void onClick(View v) {
            itemCallback.onItemClicked(getAdapterPosition());
        }
    }
}

File: app/src/main/java/com/wolcano/musicplayer/music/ui/adapter/customdialog/CopyItemAdapter.kt
package com.wolcano.musicplayer.music.ui.adapter.customdialog

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.core.content.ContextCompat
import androidx.core.text.HtmlCompat
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.wolcano.musicplayer.music.R
import com.wolcano.musicplayer.music.data.model.Copy
import com.wolcano.musicplayer.music.databinding.ItemCopyBinding
import com.wolcano.musicplayer.music.listener.ItemCallback
import java.util.*

class CopyItemAdapter(private val context: Context, private val copyList: ArrayList<Copy>): RecyclerView.Adapter<CopyItemAdapter.ViewHolder>() {

    private var itemCallback: ItemCallback? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CopyItemAdapter.ViewHolder {
        val binding: ItemCopyBinding = DataBindingUtil.inflate(
            LayoutInflater.from(parent.context),
            R.layout.item_copy,
            parent,
            false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: CopyItemAdapter.ViewHolder, position: Int) {
        holder.binding.copy = copyList[position]
        holder.binding.executePendingBindings()
        val copy: Copy? = holder.binding.copy
        holder.binding.txtName.text = copy?.text

        if (copy?.icon == 0) holder.binding.imgIcon.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.baseline_music_note_white_24))
        if (copy?.icon == 1) holder.binding.imgIcon.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.account_circle_white))
        if (copy?.icon == 2) holder.binding.imgIcon.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.baseline_album_white_36))
    }

    override fun getItemCount(): Int {
        return copyList.size
    }

    fun setCallback(callback: ItemCallback?) {
        itemCallback = callback
    }

    inner class ViewHolder(val binding: ItemCopyBinding) : RecyclerView.ViewHolder(
        binding.root
    ),
        View.OnClickListener {
        override fun onClick(v: View) {
            itemCallback?.onItemClicked(absoluteAdapterPosition)
            val replacedStr = copyList[absoluteAdapterPosition].text.replace("\"".toRegex(), "")
            val clipboard =
                context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
            val clip =
                ClipData.newPlainText(context.getString(R.string.copy_song_infos), replacedStr)
            clipboard.primaryClip = clip
            Toast.makeText(
                context,
                HtmlCompat.fromHtml(context.getString(R.string.copy_to_clipboard, replacedStr), HtmlCompat.FROM_HTML_MODE_LEGACY),
                Toast.LENGTH_SHORT
            ).show()
        }

        init {
            binding.root.setOnClickListener(::onClick)
        }
    }

}